File: packages/javascript/lib/word/randSport.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Sports and the things people play.
 *
 * `randWord({ theme: 'sport' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randSport({ language: 'ko', count: 3 }); // ['축구', '야구', '양궁']
 * randSport({ language: 'en', count: 3 }); // ['Soccer', 'Baseball', 'Archery']
 */
export function randSport(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Sport words with the language and theme each one came from.
 *
 * @example
 * randSport({ language: 'ko', output: 'detail' });
 * // [{ word: '축구', language: 'ko', theme: 'sport' }]
 */
export function randSport(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randSport(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('sport', options);
}

File: packages/dart/lib/src/word/rand_sport.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Sports and the things people play.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.sport] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randSport(language: WordLanguage.ko, count: 3); // [축구, 야구, 양궁]
/// randSport(language: WordLanguage.en, count: 3); // [Soccer, Baseball, Archery]
/// ```
List<String> randSport({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.sport,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
